### PR TITLE
Fix some serialization function and Add Annotation

### PR DESCRIPTION
**Pull request by TrustHenry**
add pure & serialization function Annotation

### CODE REVIEW

File: source/agora/node/BlockSerialize.d [hunk 1]
```diff
@@ -18,8 +18,22 @@ import agora.common.Data;
...
+
+    Serialize the block struct and return it to ubyte[].
+
+    Params:
+      T = Type of struct to serialize
+      record = Instance of `T` to serialize
+      dg  = Serialize delegate, when this struct is nested in another.
+
+    Returns:
+      The `serialized ubyte[]`
```
Comment (Geod24) [inline source/agora/node/BlockSerialize.d]:
```suggestion
      The serialized `ubyte[]`
```

**Review by Geod24 — COMMENTED** [review]
LGTM besides one nitpick

File: source/agora/node/BlockSerialize.d [hunk 2]
```diff
@@ -18,8 +18,22 @@ import agora.common.Data;
 /// Type of delegate SerializeDg
 public alias SerializeDg = void delegate(scope const(ubyte)[]) pure nothrow @safe;
 
+/*******************************************************************************
+
+    Serialize the block struct and return it to ubyte[].
```
Comment (AndrejMitrovic) [inline source/agora/node/BlockSerialize.d]:
```suggestion
    Serialize the block struct and return it as a `ubyte[]`.
```

File: source/agora/node/BlockSerialize.d [hunk 3]
```diff
@@ -18,8 +18,22 @@ import agora.common.Data;
...
+
+    Params:
+      T = Type of struct to serialize
+      record = Instance of `T` to serialize
+      dg  = Serialize delegate, when this struct is nested in another.
```
Comment (AndrejMitrovic) [inline source/agora/node/BlockSerialize.d:28]: ~This parameter no longer exists.~ Edit: I was wrong.

Comment (Geod24) [reply source/agora/node/BlockSerialize.d:28]: It's used for `ditto`

Comment (AndrejMitrovic) [reply source/agora/node/BlockSerialize.d:28]: Please ignore my comment.

File: source/agora/node/BlockSerialize.d (hunk 2 above)
Comment (TrustHenry) [reply source/agora/node/BlockSerialize.d]: Applied the suggestion

File: source/agora/node/BlockSerialize.d (hunk 1 above)
Comment (TrustHenry) [reply source/agora/node/BlockSerialize.d]: Applied the suggestion

File: source/agora/node/BlockSerialize.d (hunk 3 above)
Comment (TrustHenry) [reply source/agora/node/BlockSerialize.d:28]: Thx,

File: source/agora/node/BlockSerialize.d (hunk 2 above)
Comment (AndrejMitrovic) [reply source/agora/node/BlockSerialize.d]: did you forget to push?

**Review by AndrejMitrovic — APPROVED** [review]

File: source/agora/node/BlockSerialize.d (hunk 2 above)
Comment (TrustHenry) [reply source/agora/node/BlockSerialize.d]: How do you know? :) Thank you for the quick review.

**Pull request merged**